data/data-pipeline/data_pipeline/etl/score/etl_score.py

```diff
@@ -544,9 +470,129 @@ def transform(self) -> None:
             # )
         )
 
-        self.df["Score F (communities)"] = (
-            self.df[meets_socio_field_name] & self.df[meets_burden_field_name]
+        df["Score F (communities)"] = (
+            df[meets_socio_field_name] & df[meets_burden_field_name]
         )
+        return df
+
+    def _prepare_initial_df(self, data_sets: list) -> pd.DataFrame:
+        logger.info("Preparing initial dataframe")
+
+        # Join all the data sources that use census block groups
+        census_block_group_dfs = [
+            self.ejscreen_df, 
+            self.census_df, 
+            self.housing_and_transportation_df,
+        ]
+        census_block_group_df = self._join_cbg_dfs(census_block_group_dfs)
+
+        # Join all the data sources that use census tracts
+        census_tract_dfs = [
+            self.hud_housing_df,
+            self.cdc_places_df,
+        ]
+        census_tract_df = self._join_tract_dfs(census_tract_dfs)
+
+        # Calculate the tract for the CBG data.
+        census_block_group_df[
+            self.GEOID_TRACT_FIELD_NAME
+        ] = census_block_group_df[self.GEOID_FIELD_NAME].str[0:11]
+
+        df = census_block_group_df.merge(
+            census_tract_df, on=self.GEOID_TRACT_FIELD_NAME
+        )
+
+        if len(census_block_group_df) > 220333:
+            raise ValueError("Too many rows in the join.")
```

Comment:
If GEOID10s are read as numbers instead of strings, the initial 0 is dropped, and then we get too many CBG rows (one for 012345 and one for 12345).

Reply:
^ put as comment

Reply:
are there other errors we should be raising?

Reply:
I think this should be a type check on the ETL classes and the score should assume the right inputs